SofaGLFW/src/SofaGLFW/SofaGLFWBaseGUI.h

```diff
@@ -29,117 +30,128 @@
-
-    GLFWmonitor* getCurrentMonitor(GLFWwindow *window);
-
-    bool isFullScreen(GLFWwindow* glfwWindow = nullptr) const;
-    void switchFullScreen(GLFWwindow* glfwWindow = nullptr, unsigned int /* screenID */ = 0);
-    void setBackgroundColor(const sofa::type::RGBAColor& newColor, unsigned int /* windowID */ = 0);
-    void setBackgroundImage(const std::string& /* filename */, unsigned int /* windowID */ = 0);
-
-    sofa::core::sptr<sofa::simulation::Node> getRootNode() const;
-    bool hasWindow() const { return m_firstWindow != nullptr; }
-
-    [[nodiscard]] std::string getFilename() const
-    {
-        return m_filename;
-    }
-
-    sofa::component::visual::BaseCamera::SPtr findCamera(sofa::simulation::NodeSPtr groot);
-    void changeCamera(sofa::component::visual::BaseCamera::SPtr newCamera);
-    void setWindowIcon(GLFWwindow* glfwWindow);
-
-    void setGUIEngine(std::shared_ptr<BaseGUIEngine> guiEngine)
-    {
-        m_guiEngine = guiEngine;
-    }
-    
-    std::shared_ptr<BaseGUIEngine> getGUIEngine()
-    {
-        return m_guiEngine;
-    }
-    
-private:
-    // GLFW callbacks
-    static void error_callback(int error, const char* description);
-    static void key_callback(GLFWwindow* window, int key, int scancode, int action, int mods);
-    static void cursor_position_callback(GLFWwindow* window, double xpos, double ypos);
-    static void mouse_button_callback(GLFWwindow* window, int button, int action, int mods);
-    static void scroll_callback(GLFWwindow* window, double xoffset, double yoffset);
-    static void close_callback(GLFWwindow* window);
-    // empty (as in non-implemented) GLFW callbacks
-    static void window_focus_callback(GLFWwindow* window, int focused);
-    static void cursor_enter_callback(GLFWwindow* window, int entered);
-    static void monitor_callback(GLFWmonitor* monitor, int event);
-    static void character_callback(GLFWwindow* window, unsigned int codepoint);
-
-
-    static int handleArrowKeys(int key);
-
-    void makeCurrentContext(GLFWwindow* sofaWindow);
-    void runStep();
-
-    // static members
-    inline static std::map< GLFWwindow*, SofaGLFWWindow*> s_mapWindows{};
-    inline static std::map< GLFWwindow*, SofaGLFWBaseGUI*> s_mapGUIs{};
-
-    //members
-    bool m_bGlfwIsInitialized{ false };
-    bool m_bGlewIsInitialized{ false };
-
-    sofa::simulation::NodeSPtr m_groot;
-    std::string m_filename;
-    sofa::gl::DrawToolGL* m_glDrawTool{ nullptr };
-    sofa::core::visual::VisualParams* m_vparams{ nullptr };
-    GLFWwindow* m_firstWindow{ nullptr };
-    int m_windowWidth{ 0 };
-    int m_windowHeight{ 0 };
-    int m_lastWindowPositionX{ 0 };
-    int m_lastWindowPositionY{ 0 };
-    int m_lastWindowWidth{ 0 };
-    int m_lastWindowHeight{ 0 };
-
-    bool m_isMouseInteractionEnabled { false };
-    
-    std::shared_ptr<sofaglfw::BaseGUIEngine> m_guiEngine;
-
-};
-
-} // namespace sofaglfw
+    class SofaGLFWMouseManager;
```

Comment:
Can you revert to the previous indentation style? It's difficult to review the changes if you re-format a lot of code.